### PR TITLE
TRIO-9 TRIO-11 Feature: drag and drop functionality

### PR DESCRIPTION
**Pull request by Alex-AK**
Ready for review.

**NOTE:** This PR does not include functionality to move cards to different columns.

Add drag and drop dependency.
Add drag and drop functionality for listing cards moved within a column.
Add utility function to calculate new listing card position.
Add position property to listing POST request.

### CODE REVIEW

**Comment by nateboyette** [comment]
Rebased on master

File: .flowconfig [hunk 1]
```diff
@@ -1,7 +1,8 @@
 [ignore]
-front-end/node_modules/.*
+.*/node_modules/.*
```
Comment (nateboyette) [inline .flowconfig]: I'm getting `Cannot resolve module `react-beautiful-dnd` inside of `Column` on the `Droppable` import. Not sure if it's related to this. I tried updated flow imports and even reverting back but that didn't clear the error. Can look into this later.

File: src/utils/calculateNewPosition.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  // card dropped at last position
+  else if (destination.index === listings.length - 1) {
+    newPosition = listings[listings.length - 1].position + 1;
+  } else {
+    // card dropped somewhere else -- divide previous index position and next index position by 2.
+    const previousListingPosition = listings[destination.index - 1].position;
+    const nextListingPosition = listings[destination.index].position;
+
+    newPosition = (previousListingPosition + nextListingPosition) / 2;
```
Comment (nateboyette) [inline src/utils/calculateNewPosition.js]: All other cases seem to be working but dropping cards between each other within a column doesn't seem to be working. When moving a card to the next or previous position, that's NOT first or last within the column, the card will swap back to its original position. 

**Side Note: This is working on the next PR**

Comment (Alex-AK) [reply src/utils/calculateNewPosition.js]: Yeah, this was an oversight on my part. The next PR handles the cases for dragging cards up and down the same column.

File: .flowconfig (hunk 1 above)
Comment (Alex-AK) [reply .flowconfig]: I'm not seeing this issue. 
image

Comment (nateboyette) [reply .flowconfig]: Weird. Everything works but I'm Flow isn't happy.

**Pull request merged**